### PR TITLE
revoke_credential -> revoke_credentials to allow more efficient implementation

### PR DESCRIPTION
**Pull request by andrewwhitehead**
Also adds issuance_by_default parameter to create-registry endpoint
Fixes some method signatures in base classes

### CODE REVIEW

File: aries_cloudagent/protocols/issue_credential/v1_0/manager.py [hunk 1]
```diff
@@ -752,27 +752,15 @@ def context(self) -> InjectionContext:
...
+                registry_record.revoc_reg_id,
+                registry_record.tails_local_path,
+                revoke_idxs,
+            )
+            registry_record.revoc_reg_entry = json.loads(delta_json)
             await registry_record.publish_registry_entry(self.context)
-            result[registry_record.revoc_reg_id] = [
-                cr_id for cr_id in registry_record.pending_pub
-            ]
+            result[registry_record.revoc_reg_id] = revoke_idxs
```
Comment (sklump) [inline aries_cloudagent/protocols/issue_credential/v1_0/manager.py:763]: Doesn't this clobber any prior pending revocations? They won't make it into the delta, and they won't be pending anymore.

File: aries_cloudagent/revocation/models/issuer_rev_reg_record.py [hunk 2]
```diff
@@ -138,8 +138,7 @@ def _check_url(self, url) -> None:
         if self.state != IssuerRevRegRecord.STATE_INIT:
             raise RevocationError(
                 "Revocation registry {} in state {}: cannot generate".format(
-                    self.revoc_reg_id,
-                    self.state
+                    self.revoc_reg_id, self.state
```
Comment (sklump) [inline aries_cloudagent/revocation/models/issuer_rev_reg_record.py:141]: Is there any way to shut flake up about this kind of thing? I can read left-to-right, or I can read up-to-down. When you make me read both, it's a wall of text: poor choice when details matter.

File: aries_cloudagent/protocols/issue_credential/v1_0/manager.py (hunk 1 above)
Comment (andrewwhitehead) [reply aries_cloudagent/protocols/issue_credential/v1_0/manager.py:763]: I'm not sure I understand. It's doing the same thing as before, creating a combined delta from all the pending indexes, just doing that inside the IndyIssuer class instead. The credx issuer has an optimized path that creates a combined delta all at once.

File: aries_cloudagent/revocation/models/issuer_rev_reg_record.py (hunk 2 above)
Comment (andrewwhitehead) [reply aries_cloudagent/revocation/models/issuer_rev_reg_record.py:141]: It's the 'black' formatter, as in you can get any colour you want as long as it's black. :|

File: aries_cloudagent/protocols/issue_credential/v1_0/manager.py (hunk 1 above)
Comment (sklump) [reply aries_cloudagent/protocols/issue_credential/v1_0/manager.py:763]: Oh - I missed a line. It's fine.

**Comment by sklump** [comment]
Please merge ASAP, I have a change or two in this space that I don't want to merge with it manually.

File: aries_cloudagent/issuer/indy.py [hunk 3]
```diff
@@ -234,29 +234,57 @@ def make_credential_definition_id(
...
 
-        return revoc_reg_delta_json
+        result_json = None
+        for cred_revoc_id in cred_revoc_ids:
+            with IndyErrorHandler("Exception when revoking credential", IssuerError):
+                # may throw AnoncredsInvalidUserRevocId if using ISSUANCE_ON_DEMAND
+                delta_json = await indy.anoncreds.issuer_revoke_credential(
+                    self.wallet.handle, tails_reader_handle, revoc_reg_id, cred_revoc_id
+                )
+                if not result_json:
```
Comment (sklump) [inline aries_cloudagent/issuer/indy.py:261]: My two cents: if not ... else is a sign of overthinking.

Comment (andrewwhitehead) [reply aries_cloudagent/issuer/indy.py:261]: Maybe underthinking..

**Pull request merged**